### ISSUE TITLE
Add support for $meta projections and sorting for textScore metadata [DATAMONGO-968]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-data-mongodb#1891]
**[A Mare](https://jira.spring.io/secure/ViewProfile.jspa?name=amare)** opened **[DATAMONGO-968](https://jira.spring.io/browse/DATAMONGO-968?redirect=false)** and commented

The $meta projection operator was introduced in MongoDB 2.6. It can be used to include the textScore metadata field in the returned document. It can also be part of a sort expression.
Currently, there is no support in the framework for this kind of operations.
I tried composing a query like this:

```java
DBObject scoreProjection = new BasicDBObject("score", "{ $meta: \"textScore\" }");
Query query = new BasicQuery("{}", scoreProjection);
```

and passed it to `MongoTemplate`'s `find` method, but there are no fields in the returned document object. If we apply the `$meta` operator in a query in the mongo shell, it appends the projected field to all the fields of the document.
To summarize, I am asking for the possibility to build a query like:

```javascript
db.articles.find(
   { $text: { $search: "cake" } },
   { score: { $meta: "textScore" } }
).sort( { score: { $meta: "textScore" } } )
```

using `Query`, `Criteria`, `Field` classes.
This improvement would be very useful when performing text searches, because you will usually want the documents sorted by the textScore metadata

---

**Reference URL:** http://docs.mongodb.org/manual/reference/operator/projection/meta/

**Issue Links:**
- [DATAMONGO-850](https://jira.spring.io/browse/DATAMONGO-850) Add support for text search using $text
